source/_docs/automation/trigger.markdown

```diff
@@ -257,7 +257,7 @@ automation:
 
 {% endraw %}
 
-More dynamic and complex calculations can be done with `value_template`.
+More dynamic and complex calculations can be done with `value_template`. When this option is specified an object named `state` will be available which is populated with the state object of the entity specified by `entity_id`. See the [state objects documentation](/docs/configuration/state_object/) for details on how to access the contents of this object. 
```

Comment:
To mirror documentation about similar variables you might want to go with:

```
More dynamic and complex calculations can be done with `value_template`. The variable 'state' is the [state object](https://www.home-assistant.io/docs/configuration/state_object) of the entity specified by `entity_id`.
```

It would also be helpful to add or modify one of the example configurations to include the use of `state.state`.

Reply:
Updated with the more concise explanation, and also added and example with `state.state`